pkg/controllers/dashboard/hostedcluster/controller.go

```diff
@@ -134,6 +138,23 @@ func (h handler) onClusterChange(key string, cluster *v3.Cluster) (*v3.Cluster,
 	return cluster, nil
 }
 
+// check helm release secrets for aks/eks/gke operator chart, if it has been uninstalled, removed it in m.manager.desiredChart
+// so that we don't automatically redeploy it unless there is an AKS/EKS/GKE cluster trigerring it
+func (h handler) onSecretChange(key string, obj *corev1.Secret) (*corev1.Secret, error) {
+	if obj == nil {
+		ns, name := kv.Split(key, "/")
+		if ns == namespace.System {
+			// the name will follow the format sh.helm.release.v1.rancher-eks-operator-crd.v1
+			parts := strings.Split(name, ".")
+			if len(parts) == 6 {
```

Comment:
Couldn't this accidentally match other things? Maybe this could be changed to a regex match to match on `/sh.helm.release.v1.rancher-.*-operator(-crd)?.v1/`

Reply:
Oh you are right, regex might be overkill. I can do a name matching.